apps/emqx_rule_engine/src/emqx_rule_runtime.erl

```diff
@@ -394,38 +394,36 @@ eval({'case', CaseOn, CaseClauses, ElseClauses}, Columns) ->
 eval({'fun', {_, Name}, Args}, Columns) ->
     apply_func(Name, [eval(Arg, Columns) || Arg <- Args], Columns).
 
-handle_alias({path, [{key, <<"payload">>} | _]}, #{payload := Payload} = Columns) ->
-    Columns#{payload => may_decode_payload(Payload)};
-handle_alias({path, [{key, <<"payload">>} | _]}, #{<<"payload">> := Payload} = Columns) ->
-    Columns#{<<"payload">> => may_decode_payload(Payload)};
-handle_alias(_, Columns) ->
-    Columns.
-
-alias({var, Var}) ->
+alias({var, Var}, _Columns) ->
     {var, Var};
-alias({const, Val}) when is_binary(Val) ->
+alias({const, Val}, _Columns) when is_binary(Val) ->
     {var, Val};
-alias({list, L}) ->
+alias({list, L}, _Columns) ->
     {var, ?ephemeral_alias(list, length(L))};
-alias({range, R}) ->
+alias({range, R}, _Columns) ->
     {var, ?ephemeral_alias(range, R)};
-alias({get_range, _, {var, Key}}) ->
+alias({get_range, _, {var, Key}}, _Columns) ->
     {var, Key};
-alias({get_range, _, {path, Path}}) ->
-    {path, Path};
-alias({path, Path}) ->
-    {path, Path};
-alias({const, Val}) ->
+alias({get_range, _, {path, _Path} = Path}, Columns) ->
+    handle_path_alias(Path, Columns);
+alias({path, _Path} = Path, Columns) ->
+    handle_path_alias(Path, Columns);
+alias({const, Val}, _Columns) ->
     {var, ?ephemeral_alias(const, Val)};
-alias({Op, _L, _R}) when ?is_arith(Op); ?is_comp(Op) ->
+alias({Op, _L, _R}, _Columns) when ?is_arith(Op); ?is_comp(Op) ->
     {var, ?ephemeral_alias(op, Op)};
-alias({'case', On, _, _}) ->
+alias({'case', On, _, _}, _Columns) ->
     {var, ?ephemeral_alias('case', On)};
-alias({'fun', Name, _}) ->
+alias({'fun', Name, _}, _Columns) ->
     {var, ?ephemeral_alias('fun', Name)};
-alias(_) ->
+alias(_, _Columns) ->
     ?ephemeral_alias(unknown, unknown).
 
+handle_path_alias({path, [{key, <<"payload">>} | Rest]}, #{payload := _Payload} = _Columns) ->
```

Comment:
make sure the `payload` in the path has the same type as it in the columns or it possibly be duplicated